test/BuildTools.Tasks.Tests/UnzipArchiveTest.cs

```diff
@@ -54,6 +54,30 @@ public void UnzipsFile()
             }
         }
 
+        [Fact]
+        public void Overwrites()
+        {
+            var files = new[]
+            {
+                "a.txt",
+                "dir/b.txt"
+            };
+
+            var dest =CreateZip(files);
+            var outDir = Path.Combine(_tempDir, "out");
+
+            var task = new UnzipArchive
+            {
+                File = dest,
+                Destination = outDir,
+                BuildEngine = new MockEngine(),
+                Overwrite = true
+            };
+
+            Assert.True(task.Execute(), "Task failed");
+            Assert.True(task.Execute(), "Task failed on second run.");
```

Comment:
Would be good to assert it actually wrote to the file. Before you run the task, create `$outDir/a.txt` with contents. Assert it is overwritten as an empty file after the task runs.